### PR TITLE
Combat: real random crit/landing/proc draws (replace deterministic accumulator)

### PR DESCRIPTION
**Pull request by TheSusort**
## Summary

Replaces the combat engine's deterministic fractional-accumulator gate with **true `Math.random()` draws** for crit, debuff landing (hacking vs security), charge manipulation, and proc chances — in both the combat simulator and the DPS/healing calculators (they share the same `runCombat` engine).

The old accumulator was back-loaded: at 99% crit the **first hit could never crit** (it fired every-other-hit on a fixed schedule). Draws are now real random samples (`rng() < rate`), so a 99% rate crits ~99% of the time with natural variance.

### Production change (minimal)
- `src/utils/calculators/rateAccumulator.ts`: `makeRateGate()` now returns `(rate) => rng() < clamp(rate, 0, 1)`, where module-level `rng` defaults to `Math.random`. `rollRateGate` inherits randomness for free (it delegates to `makeRateGate`); zero call-site churn. Edge cases preserved: `rate >= 1` always fires, `rate <= 0` never fires.
- Added **test-only** `setRateGateRng` / `resetRateGateRng` + a `mulberry32` PRNG. Production never seeds — it stays truly random.

### Test seam
- `src/setupTests.ts` installs a seeded `mulberry32` per test (global `beforeEach`/`afterEach`), so the suite — including golden snapshots — stays deterministic without seeding production.
- Golden snapshots (`dpsGoldenParity`, `healingGoldenParity`) re-baselined once under the seeded sequence; the crit-0/100 (RNG-independent) anchors were verified **unchanged**, confirming the flip didn't alter always/never-fire behavior.
- Schedule-coupled tests had their RNG forced to recover original intent (assertions unchanged); draw-order-coupled sequences are flagged `ORDER-SENSITIVE`.

### User-facing
- Per the design, **no Monte Carlo / no product seed** (YAGNI). The DPS calculator's headline figure now varies slightly between runs — called out in the changelog.

Spec: `docs/superpowers/specs/2026-06-30-random-crit-checks-design.md`
Plan: `docs/superpowers/plans/2026-06-30-random-crit-checks.md`

## Test Plan
- [x] `npm test` — 278 files / 3683 tests pass, 0 failures
- [x] `npm run lint` — clean (max-warnings 0)
- [x] Golden re-baseline diffs are reordering-only; crit-0/100 anchors unchanged
- [ ] Manual: run the combat sim / DPS calculator a few times and confirm crit outcomes vary (and the opener can crit at high crit rate)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/utils/combat/triggers.ts [hunk 1]
```diff
@@ -807,8 +807,8 @@ export interface IntentExecContext {
      *  key makes cross-side collisions impossible by construction. */
     repairCountBySource?: Map<string, number>;
     /** Combat-lifetime per-ability proc-chance gates (e.g. Bloodthirst's 12% chance).
-     *  Keyed `${ownerId}:${abilityId}`; the RateGate accumulates across all rounds and all
-     *  reactive fires of the same ability so the proc lands at its true frequency. */
+     *  Keyed `${ownerId}:${abilityId}`; the RateGate fires with the proc's probability on
+     *  each reactive draw of the same ability so the proc lands at its true frequency. */
```
[the remark this answers is not in the document]
Comment (TheSusort) [reply src/utils/combat/triggers.ts:811]: Not changing this — the `pc <= 0 → return true` boundary is an intentional, long-standing pass-through contract, not a bug introduced here.

`passesProcChanceGate`'s own JSDoc (lines 1257–1262) documents it: "Pass-through when procChance is undefined / <=0 / >=1". In the data model an ability either specifies a `procChance` in the open interval (0,1) or omits it; `0` and `1` are treated as ungated (always fire), the same as absent. This is locked by an existing test — `procChance exactly 0: gate bypassed, fires on every trigger (treated as absent)` — which asserts all 10 triggers fire.

Crucially, `pc <= 0` returns `true` *before* reaching `makeRateGate()`, so `makeRateGate`'s rate≤0 boundary is never exercised here — the two layers have independent contracts by design, nothing is broken. This PR only edited the field doc-comment, not this guard.
